scripts/babel/wrap-warning-with-dev-check.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Babel plugin that wraps `warning` calls with development check to be
+ * completely stripped from the production bundle.
+ *
+ * In the development bundle, warnings get wrapped with their condition
+ * and their condition becomes false to trigger them without evaluating twice.
+ *
+ * Input:
+ *
+ * ```
+ * warning(condition, message);
+ * ```
+ *
+ * Output:
+ *
+ * ```
+ * if (__DEV__) {
+ *   if (!condition) {
+ *     warning(false, message);
```

Comment:
```suggestion
 *     warning(condition, message);
```

could this not be sufficient in output?

Reply:
WDYM?

In development, we want the reverse of the condition to be in `if` and the condition in warning to be `false`. Remember: the warning message is displayed if the condition evaluates to `false` (that can be confusing and we might want to change that at some point).

Reply:
I meant that this should be enough:

```
warning(condition, message);

// becomes

__DEV__ ? warning(condition, message) : void 0;

// instead of

__DEV__ ? !condition ? warning(false, message) : void 0 : void 0;
```

Not sure why that extra wrapper is necessary :)

Reply:
That should indeed be enough in our case, thanks!

Reply:
See 656b44da1f956909853662a6fe38137a46063695.